agent/api/container.go

```diff
@@ -128,11 +128,17 @@ type Container struct {
 	// is handled properly so that the state storage continues to work.
 	KnownStatusUnsafe ContainerStatus `json:"KnownStatus"`
 
+	// TransitionDependenciesMap is a map of the dependent container status to other
+	// dependencies that must be satisfied in order for this container to transition.
+	TransitionDependenciesMap map[ContainerStatus]TransitionDependencySet
+
 	// TransitionDependencySet is a set of dependencies that must be satisfied
 	// in order for this container to transition.  Each transition dependency
 	// specifies a resource upon which the transition is dependent, a status
 	// that depends on the resource, and the state of the dependency that
 	// satisfies.
+	// Deprecated: Use TransitionDependenciesMap instead. TransitionDependencySet is
```

Comment:
A better way to do this (in your future PR) would be to:

1. get rid of  `TransitionDependencySet` altogether
2. tag `TransitionDependenciesMap` as `"TransitionDependencySet"` in JSON
3. update the JSON marshaler to deal with `"TransitionDependencySet"` tag. If it's a map, no big deal. Else, construct the `TransitionDependenciesMap` map.

Doing that would mean that we don't need to have duplicate code, names for this.